### DISCUSSION TITLE
Upgrade MySQL 8.0.30 to 8.4.2 to 9.0.1

### DISCUSSION THREAD

**Discussion by Donny-168** [leokhoa/laragon#888] [General]
Hi...i just upgrade MySQL 8.0.30 to 9.0.1, i follow all instruction and run ok...status MySQL 9.0.1 Started...and try to change root password and success...but i can't still connect my localhost 3307...when i back to use MySQL 8.0.30...everything is ok.
What should i do to solve this problem?

![Laragon_mysql2](https://github.com/user-attachments/assets/6755f9da-1db3-474b-8100-cb036c6d06f9)

![Laragon_mysql1](https://github.com/user-attachments/assets/45f8c261-50fc-4acd-9cbf-86d3a9c4e11d)

**Comment by Donny-168** [comment]
**SOLVE**
1. **_Before upgrade, backup file in folder Laragon/Data_**
2. Edit file at folder laragon/MySql/Bin/MySQL xxxxx/My.Ini
3. Make file My.INI into "Read Only" file
4. Use HeidiSQL to connect MySQL especially MySQL 9.0 

![LARAGON](https://github.com/user-attachments/assets/2526e2bc-e7d4-4fae-b688-734b1220fb7c)

![LARAGON BIN](https://github.com/user-attachments/assets/f4249410-24cf-45e3-9804-43ff35462579)
![LARAGON DATA](https://github.com/user-attachments/assets/83db6481-94f9-45f4-9a89-5a7624cc30da)

`My.Ini 8.0.30`
![LARAGON_3307 ini](https://github.com/user-attachments/assets/c3797649-62a4-4bcd-8d8f-f83a28a4abbb)
![LARAGON_3307](https://github.com/user-attachments/assets/44719eea-30ea-404f-92c8-207574a0f5ea)

`My.Ini 8.4.2`
![LARAGON_3308 ini](https://github.com/user-attachments/assets/59ba7234-0f16-4a0b-ba8f-86ac14d805d6)
![LARAGON_3308](https://github.com/user-attachments/assets/bc1ab665-cf1c-45b7-96df-f2bb7a176164)

`My.Ini 9.0.1`
![LARAGON_3309 ini](https://github.com/user-attachments/assets/ca19920b-791e-4941-9cd8-439ec8539039)
![LARAGON_3309](https://github.com/user-attachments/assets/b87cf6ad-b440-4f91-9787-3a3735d2b76a)

`Read Only`
![LARAGON REA ONLY](https://github.com/user-attachments/assets/bd346fe7-bbaf-4e0e-9574-37a7c33ddbfa)

**Comment by Ringoinmel** [comment]
I got this error.  Any idea?  Thanks

2024-11-16T12:11:50.318872Z 0 [ERROR] [MY-000067] [Server] unknown variable 'default_authentication_plugin=mysql_native_password'.
2024-11-16T12:11:50.323694Z 0 [ERROR] [MY-010119] [Server] Aborting

**Comment by Ringoinmel** [comment]
fixed and now I got SQL version 9.1.0

**Reply by Donny-168** [reply]
> fixed and now I got SQL version 9.1.0

Hi, can u share how to configure My SQL ver 9.1.0?...

**Reply by rpascolo** [reply]
Follow instructions to move to 8.4.2 : https://gist.github.com/meorajrul/b57803bf1b4ddfd2f93e6ad37c3ac5f2
Instructions are the same for 9.1 (adapt datadir and basedir to match your configuration)

**Reply by Donny-168** [reply]
> Follow instructions to move to 8.4.2 : https://gist.github.com/meorajrul/b57803bf1b4ddfd2f93e6ad37c3ac5f2 Instructions are the same for 9.1 (adapt datadir and basedir to match your configuration)

![LARAGON_3310](https://github.com/user-attachments/assets/8f980ee2-cd1a-4c1a-bbbe-d0bae1ce542f)

**Reply by Ringoinmel** [reply]
my.ini as below for SQL 9.1.0 and also make it read only

[client]
#password=your_password
port=3306
socket=/tmp/mysql.sock

[mysqld]
port=3306
socket=/tmp/mysql.sock
key_buffer_size=256M
max_allowed_packet=512M
table_open_cache=256
sort_buffer_size=1M
read_buffer_size=1M
read_rnd_buffer_size=4M
myisam_sort_buffer_size=64M
thread_cache_size=8

secure-file-priv=""
explicit_defaults_for_timestamp=1
datadir= "C:/laragon/data/mysql-9.1"
#default_authentication_plugin=mysql_native_password


[mysqldump]
quick
max_allowed_packet=512M
